src/webdav-server/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,108 @@
+import { CryptoProvider, LoginDetails } from '@internxt/sdk';
+import { Keys, Password } from '@internxt/sdk/dist/auth';
+import { UserSettings } from '@internxt/sdk/dist/shared/types/userSettings.js';
+import { aes } from '@internxt/lib';
+import { SdkManager } from '../../../core/SDKManager.ts';
+import { decryptText, decryptTextWithKey, encryptText, passToHash } from '../../crypto/services/utils.ts';
+import { generateNewKeys } from '../../crypto/services/pgp.service.ts';
+import {
+  assertPrivateKeyIsValid,
+  assertValidateKeys,
+  decryptPrivateKey,
+  getAesInitFromEnv,
+} from '../../crypto/services/keys.service.ts';
+
+const generateNewKeysWithEncrypted = async (password: string) => {
```

Comment:
I'm a fan of functional style programming, but I would say either we stick to functional style services, or singleton based services that are easy to test (you can always mock dependencies there + you have the benefits of OOP). Would you agree to stick to this services organization pattern for better consistency?

https://github.com/internxt/cli/blob/main/src/services/config.service.ts

Basically you have a class, the class always have an instance ready, and if you want to test/mock the class, you can use `new YourClass` in a test